src/metabase/models/permissions/parse.clj

```diff
@@ -110,8 +124,8 @@
                                      [:db db-id :data-model :schemas :all])
     [:dm-db db-id db-node]         (let [db-id (Long/parseUnsignedLong db-id)]
                                      (into [:db db-id :data-model :schemas] (path2 db-node)))
-    [:dm-schema schema-name]       [schema-name :all]
-    [:dm-schema schema-name table] (into [schema-name] (path2 table))
+    [:dm-schema schema-name]       [(path1 schema-name) :all]
+    [:dm-schema schema-name table] (into [(path1 schema-name)] (path2 table))
```

Comment:
Not a huge deal, but I think it's easier to read & reason about these two matching functions if each one is fully self-contained. Could you add the `:schema-name` clause to this one as well, and then call `path2` here instead?